media-source/mediasource-activesourcebuffers.html

```diff
@@ -0,0 +1,212 @@
+                        "second SourceBuffer is the video one");
+
+                      test.expectEvent(mediaElement, "loadedmetadata");
+                      test.expectEvent(mediaSource.activeSourceBuffers, "addsourcebuffer");
+                      sourceBufferAudio.appendBuffer(dataAudio);
+                      sourceBufferVideo.appendBuffer(dataVideo);
+
+                      test.waitForExpectedEvents(function()
+                      {
+                          assert_equals(mediaSource.activeSourceBuffers.length, 2,
+                            "activeSourceBuffers list updated when tracks are loaded");
+                          assert_equals(mediaSource.activeSourceBuffers[0], sourceBufferAudio,
+                            "first active SourceBuffer is the audio one");
+                          assert_equals(mediaSource.activeSourceBuffers[1], sourceBufferVideo,
+                            "second active SourceBuffer is the video one");
+                          test.done();
+                      });
+                  });
+              });
+          }, "Active SourceBuffers must appear in the same order as they appear in the sourceBuffers attribute.");
+
+
+          mediasource_test(function(test, mediaElement, mediaSource)
+          {
+              mediaElement.addEventListener("error", test.unreached_func("Unexpected event 'error'"));
+              MediaSourceUtil.fetchManifestAndData(test, manifestFilenameAudio, function (typeAudio, dataAudio)
+              {
+                  MediaSourceUtil.fetchManifestAndData(test, manifestFilenameVideo, function (typeVideo, dataVideo)
+                  {
+                      var sourceBufferAudio = mediaSource.addSourceBuffer(typeAudio);
+                      var sourceBufferVideo = mediaSource.addSourceBuffer(typeVideo);
+
+                      test.expectEvent(sourceBufferAudio.audioTracks, "addtrack");
+                      test.expectEvent(sourceBufferVideo.videoTracks, "addtrack");
+                      sourceBufferAudio.appendBuffer(dataAudio);
+                      sourceBufferVideo.appendBuffer(dataVideo);
+
+                      test.waitForExpectedEvents(function()
+                      {
+                          assert_equals(mediaSource.activeSourceBuffers.length, 2,
+                            "activeSourceBuffers list updated when tracks are loaded");
+                          assert_equals(sourceBufferAudio.audioTracks.length, 1,
+                            "audio track list contains loaded audio track");
+                          assert_equals(sourceBufferVideo.videoTracks.length, 1,
+                            "video track list contains loaded video track");
+
+                          test.expectEvent(mediaSource.activeSourceBuffers, "removesourcebuffer");
+                          sourceBufferAudio.audioTracks[0].enabled = false;
+
+                          test.waitForExpectedEvents(function()
```

Comment:
The way the expectation handler is engineered, I don't think these waitForExpectedEvents need to be nested within a previous one's callback here. The utility methods around expectation handling do this.